client/layout/guided-tours/config-elements/step.js

```diff
@@ -159,7 +159,20 @@ export default class Step extends Component {
 			}
 		}
 
-		return Promise.resolve();
+		return new Promise( resolve => {
+			if ( document.querySelector( props.target ) ) {
+				return resolve();
+			}
+
+			const waitForElement = () => {
+				if ( document.querySelector( props.target ) ) {
+					return resolve();
+				}
+				setTimeout( waitForElement, 125 );
+			};
+
+			waitForElement();
```

Comment:
Good notes on a slightly simpler implementation: https://github.com/Automattic/wp-calypso/pull/33359#discussion_r288422266

Reply:
4ddd13e

Reply:
There's been further discussion: https://github.com/Automattic/wp-calypso/pull/33359#discussion_r288487000

It _probably_ isn't too relevant for that one-off soon-to-be-removed solution in #33359, but it would be nice to get it _really right_ at the framework level here.